controllers/mongodb_tls.go

```diff
@@ -329,3 +358,21 @@ func buildTLSPrometheus(mdb mdbv1.MongoDBCommunity) podtemplatespec.Modification
 		podtemplatespec.WithVolumeMounts(construct.MongodbName, tlsSecretVolumeMount),
 	)
 }
+
+func buildAgentX509(mdb mdbv1.MongoDBCommunity) podtemplatespec.Modification {
+	if mdb.Spec.GetAgentAuthMode() != "X509" {
+		return podtemplatespec.Apply(
+			podtemplatespec.RemoveVolume(constants.AgentPemFile),
```

Comment:
Could this be an optional volume instead?

Reply:
I tested with optional volumes. However what happens is that the volume still remains in the sts, but the certificate is, of course, not present in the agent container. Is this what we want, or should the volume not exist anyways?